### PR TITLE
Reset connection handling channel when firewall handler is stopped

### PR DESCRIPTION
**Pull request by dhaavi**

### CODE REVIEW

File: network/connection.go [hunk 1]
```diff
@@ -590,9 +591,15 @@ func (conn *Connection) SetFirewallHandler(handler FirewallHandler) {
 }
 
 // StopFirewallHandler unsets the firewall handler and stops the handler worker.
+// The caller needs to hold a lock on the connection.
 func (conn *Connection) StopFirewallHandler() {
 	conn.firewallHandler = nil
+
+	// Signal the packet handler worker that it can stop.
 	conn.pktQueue <- nil
+
```
Comment (ppacher) [inline network/connection.go:600]: just close the channel here and use `pkt, ok := <-pktQueue`

**Review by ppacher — CHANGES_REQUESTED** [review]

File: network/connection.go (hunk 1 above)
Comment (dhaavi) [reply network/connection.go:600]: True. But I don't have to use the `ok` syntax, as it will return the zero value anyway, right?

Comment (ppacher) [reply network/connection.go:600]: Right

**Review by ppacher — COMMENTED** [review]
LGTM

**Pull request merged**